File: Python/simple-factory/simple03/PersonFactory.py
# -*- coding: utf-8 -*-
from Man import Man
from Woman import Woman
from Robot import Robot


class PersonFactory(object):
    @staticmethod
    def producePerson(obj):
        if obj.lower() == 'man':
            print("正在生产一个男人...")
            return Man()
        elif obj.lower() == 'woman':
            print("正在生产一个女人...")
            return Woman()
        elif obj.lower() == 'robot':
            print("正在生产一个机器人...")
            return Robot()
        else:
            print(obj + "无法生产...\n")
            return None

File: Homework/Simple Factory/ch4/Nvwa/PersonFactory.java
package ch4.Nvwa;

public class PersonFactory {
	public static Person produceperson(String p){
		if(p == "W"){
			return (new Woman());
		}
		else if(p == "M"){
			return (new Man());
		}
		else if(p == "R"){
			return (new Robot());
		}
		return null;
	}
}
